tests/tests_app/core/test_queues.py

```diff
@@ -235,3 +241,23 @@ def test_unreachable_queue(monkeypatch):
     # Test backoff on queue.put
     test_queue.put("foo")
     assert test_queue.client.post.call_count == 3
+
+
+def test_rate_limited_queue():
```

Comment:
It might be good to mock the time.sleep() call  in the `_wait_until_allowed` method, to make the unit test fast.

Reply:
Done :smiley: